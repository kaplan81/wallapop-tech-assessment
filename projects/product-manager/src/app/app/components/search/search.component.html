<form class="search" [formGroup]="searchForm" (ngSubmit)="onSubmit()">
  <input
    class="search__input"
    type="text"
    placeholder="Any search gets all products"
    formControlName="searchCriteria"
  />
</form>
