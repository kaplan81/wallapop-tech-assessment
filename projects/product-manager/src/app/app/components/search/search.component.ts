import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { NonNullableFormBuilder, ReactiveFormsModule } from '@angular/forms';
import { Router } from '@angular/router';
import { ProductsStateService } from '../../services/products-state/products-state.service';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [ReactiveFormsModule],
  selector: 'mng-search',
  standalone: true,
  styleUrls: ['./search.component.scss'],
  templateUrl: './search.component.html',
})
export class SearchComponent {
  #fb = inject(NonNullableFormBuilder);
  #productsStateService = inject(ProductsStateService);
  #router = inject(Router);
  searchForm = this.#fb.group({
    searchCriteria: [''],
  });

  onSubmit(): void {
    this.#router.navigate(['products'], { queryParams: { page: 1 } }).then(() => {
      this.searchForm.reset();
      this.#productsStateService.updateStateProp('loaded', false);
    });
  }
}
