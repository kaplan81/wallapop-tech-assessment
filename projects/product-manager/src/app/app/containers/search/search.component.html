<form class="search" [formGroup]="searchForm">
  <input
    class="search__input"
    type="text"
    placeholder="Any search gets all products"
    formControlName="searchCriteria"
  />
</form>
