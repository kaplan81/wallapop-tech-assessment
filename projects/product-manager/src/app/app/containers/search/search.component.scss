.search {
  display: flex;
  justify-content: center;
  height: 50px;
  margin: 30px;
  &__input {
    font-family: Roboto, 'Helvetica Neue', sans-serif;
    font-size: 16px;
    width: 400px;
    height: 50px;
    border: 2px solid #ccc;
    border-radius: 5px;
    padding: 10px 20px;
    &:hover,
    &:focus {
      border: 2px solid #555;
      background-color: #fff;
    }
    transition: 250ms;
  }
}
