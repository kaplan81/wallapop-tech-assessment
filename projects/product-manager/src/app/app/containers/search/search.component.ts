import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { NonNullableFormBuilder, ReactiveFormsModule } from '@angular/forms';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [ReactiveFormsModule],
  selector: 'mng-search',
  standalone: true,
  styleUrls: ['./search.component.scss'],
  templateUrl: './search.component.html',
})
export class SearchComponent {
  #fb = inject(NonNullableFormBuilder);
  searchForm = this.#fb.group({
    searchCriteria: [''],
  });

  onSubmit(event: SubmitEvent): void {
    console.log('onSubmit', event);
    this.searchForm.reset();
  }
}
