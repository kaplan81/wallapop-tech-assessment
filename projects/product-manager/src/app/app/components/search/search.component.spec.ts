import { ComponentFixture, TestBed } from '@angular/core/testing';
import { SearchComponent } from './search.component';

describe('SearchComponent', () => {
  let fixture: ComponentFixture<SearchComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [SearchComponent],
    }).compileComponents();
    fixture = TestBed.createComponent(SearchComponent);
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
