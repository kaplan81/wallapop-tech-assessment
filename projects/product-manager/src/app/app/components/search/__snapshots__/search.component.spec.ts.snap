// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent should match snapshot 1`] = `
<mng-search
  __ngContext__="0"
  searchForm={[Function FormGroup]}
>
  <form
    class="search ng-untouched ng-pristine ng-valid"
    novalidate=""
  >
    <input
      class="search__input ng-untouched ng-pristine ng-valid"
      formcontrolname="searchCriteria"
      placeholder="Any search gets all products"
      type="text"
    />
  </form>
</mng-search>
`;
